File: c/muse/algorithms/sorting/InsertionSortTest.c
#include <muse/algorithms/sorting/InsertionSort.h>
#include <muse/util/SequenceBuilder.h>
#include <muse/util/Sequences.h>
#include <muse/util/TestRunner.h>

bool testInsertionSort(void) {
  size_t size = 32768;

  int arr[size];
  SequenceBuilder_packRandom(arr, size);

  int checksum = Sequences_parityChecksum(arr, size);

  InsertionSort_sort(arr, size);

  if (Sequences_parityChecksum(arr, size) != checksum) {
    return false;
  }

  return Sequences_isSorted(arr, size);
}

int main(void) { TestRunner_pick(&testInsertionSort); }

File: python/muse/algorithms/sorting/insertion_sort_test.py
from muse.algorithms.sorting import insertion_sort, tests
from muse.util import test_runner


def test_insertion_sort() -> bool:
    sort = insertion_sort.sort
    return (
        tests.derive_decreasing(sort, 10000)()
        and tests.derive_empty(sort)()
        and tests.derive_identical(sort, 100000)()
        and tests.derive_increasing(sort, 100000)()
        and tests.derive_random(sort, 10000)()
    )


def main() -> None:
    test_runner.pick(test_insertion_sort)


if __name__ == "__main__":
    main()
